### PR TITLE
4337: Add `safeOperation` inside of `userOperation` serializer

### PR DESCRIPTION
**Pull request by Uxio0**
- `user-operations` endpoints should return `safeOperation` if there's one for the `userOperation`
- Related to #2035

### CODE REVIEW

File: safe_transaction_service/history/serializers.py [hunk 1]
```diff
@@ -757,7 +757,9 @@ class SafeCreationInfoResponseSerializer(serializers.Serializer):
     master_copy = EthereumAddressField(allow_null=True)
     setup_data = HexadecimalField(allow_null=True)
     data_decoded = serializers.SerializerMethodField()
-    safe_operation = aa_serializers.SafeOperationResponseSerializer(allow_null=True)
+    user_operation = aa_serializers.UserOperationWithSafeOperationResponseSerializer(
```
Comment (moisses89) [inline safe_transaction_service/history/serializers.py:760]: This change is not indicated in the PR description, the reason behind this is that during SafeCreation the is not a SafeOperation?

**Comment by moisses89** [comment]
For me is a little bit confusing have UserOperationWithSafeOperation and SafeOperationWithUserOperation. 
What is the relation between them?

File: safe_transaction_service/history/serializers.py (hunk 1 above)
Comment (Uxio0) [reply safe_transaction_service/history/serializers.py:760]: If UserOperation is reverted there's still UserOperation and noce is increased, but there's no SafeOperation as it gets reverted

Comment (Uxio0) [reply safe_transaction_service/history/serializers.py:760]: So, a SafeOperation is always linked to a UserOperation, but the opposite is not true

**Comment by Uxio0** [comment]
> For me is a little bit confusing have UserOperationWithSafeOperation and SafeOperationWithUserOperation. What is the relation between them?

You can query a list of UserOperations (and you will get the related safeOperation if it exist) or you can query a list of SafeOperations (and you will always get the related UserOperation).

In the future we might be able to remove one of the enpoints, but currently while we figure it out how to add support for 4337 I think all the endpoints are good there

**Review by moisses89 — APPROVED** [review]

**Pull request merged**